src/hooks/useSingleStageHandler.ts

```diff
@@ -115,6 +133,14 @@ export const useSingleStageHandler = () => {
       });
       await tx.wait();
       toast.success("Your transaction was successful");
+      const obj = stage;
```

Comment:
Please check if the tx is successful or failed before updating the amount.

Reply:
If the transaction fails it will go to catch block. We are doing transactions in try and catch block to catch any error